dll/shellext/fontext/CFontCache.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ * PROJECT:     ReactOS Font Shell Extension
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     font list cache handling
+ * COPYRIGHT:   Copyright 2019 Mark Jansen (mark.jansen@reactos.org)
+ */
+
+#include "precomp.h"
+
+#define FONT_HIVE   HKEY_LOCAL_MACHINE
+#define FONT_KEY    L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Fonts"
+
+CFontCache* g_FontCache = NULL;
+
+CFontInfo::CFontInfo(LPCWSTR name)
+    : m_Name(name)
+    , m_FileRead(false)
+{
+}
+
+const CStringW& CFontInfo::Name() const
+{
+    return m_Name;
+}
+
+const bool CFontInfo::Valid() const
+{
+    return !m_Name.IsEmpty();
+}
+
+const CStringW& CFontInfo::File()
+{
+    if (!m_FileRead)
+    {
+        if (Valid())
+        {
+            CRegKey key;
+            if (key.Open(FONT_HIVE, FONT_KEY, KEY_READ) == ERROR_SUCCESS)
+            {
+                CStringW Value;
+                DWORD dwAllocated = 128;
+                LSTATUS Status;
+                do
+                {
+                    DWORD dwSize = dwAllocated;
+                    PWSTR Buffer = Value.GetBuffer(dwSize);
+                    Status = key.QueryStringValue(m_Name, Buffer, &dwSize);
+                    Value.ReleaseBuffer(dwSize);
+                    if (Status == ERROR_SUCCESS)
+                    {
+                        // Ensure we do not re-use the same string object, by passing it a PCWSTR
+                        m_File = Value.GetString();
+                        break;
+                    }
+                    dwAllocated += 128;
+                } while (Status == ERROR_MORE_DATA);
+            }
+        }
+        m_FileRead = true;
```

Comment:
Move the code `m_FileRead = true;` to the preferred position.

Reply:
It is in the preferred position.